libbeat/common/capabilities/capabilities_linux.go

```diff
@@ -0,0 +1,209 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package capabilities
+
+import (
+	"errors"
+	"math/bits"
+	"strconv"
+	"strings"
+
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+)
+
+var (
+	// Tried to map an invalid capability ID: x < 0 || x >= 64.
+	errInvalidCapability = errors.New("invalid capability")
+
+	// Capabilities are linux only and this is returned on other
+	// systems for all the public functions. There is a generic
+	// errors.ErrUnsupported present in golang 1.21, but we still
+	// support 1.20.
+	ErrUnsupported = errors.New("capabilities are only supported in linux")
+
+	// The mask when all known capabilities are set.
+	allMask = (uint64(1) << uint64(cap.MaxBits())) - 1
```

Comment:
Should this not be something like 

```go
allMask = (uint64(1) << uint64(len(ecsNames)) - 1
```
Otherwise if you are on a newer kernel with unnamed capabilities you will end up using a valid capability bit to indicate that something has every capability?

Reply:
You're right that "CAP_ALL" is host dependent, and the allMask reflects what CAP_ALL is from the host perspective.
I'm not sure I follow you, "if you are on a *newer* kernel with unnamed capabilities", did you mean an *older* kernel?
What will happen now is "if this is not my idea of CAP_ALL, make list with each individual capability, and if I don't know the name of some, call them CAP_NUMBER". Does that make sense?